### ISSUE TITLE
Allow using App Insights to query current state of orchestrations and entities

### ISSUE THREAD

**Issue by anthonychu** [Azure/azure-functions-durable-extension#1172]
Add custom dimensions to App Insights requests so that we can use the requests table to query recently run orchestrations and their current status.

Orchestrations: add instanceId and status.
Entities: add entityId and operationName.

Need to verify this is possible from the extension and that we can write the query based on those custom dimensions.

The goal is to create specialized monitor views in the portal for orchestrations and entities.

**Comment by anthonychu** [comment]
Ran some tests. Looks like this is going to work.

```csharp
using System.Diagnostics;
// ...
var activity = Activity.Current;
activity.AddTag("DurableFunctionsType", "Orchestration");
activity.AddTag("DurableFunctionsInstanceId", instanceId);
activity.AddTag("DurableFunctionsStatus", status);
```

App Insights query to pull the 30 most recently run orchestrators and their current status:

```
requests
| extend DurableFunctionsInstanceId = tostring(customDimensions["DurableFunctionsInstanceId"]),
         DurableFunctionsStatus = tostring(customDimensions["DurableFunctionsStatus"]),
         DurableFunctionsType = tostring(customDimensions["DurableFunctionsType"])
| where DurableFunctionsType == "Orchestration"
| summarize arg_max(timestamp, *) by DurableFunctionsInstanceId 
| order by timestamp desc
| limit 30
```

Thanks to @brettsam for helping with this.

**Comment by anthonychu** [comment]
@cgillum Thoughts on how this might look like for entities? Does a "status" make sense or would something else be more interesting (e.g. has state, no state)?

**Comment by cgillum** [comment]
I'm not entirely sure what would make sense for the current "status" of an entity. Maybe the size of its current state in bytes? Adding @sebastianburckhardt in case he has thoughts on this.

**Comment by sebastianburckhardt** [comment]
We currently already return selected status info in `TaskEntityShim.GetStatus`, as a JSON object. Those same fields probably would make sense to use:
```csharp
return this.dataConverter.MessageConverter.Serialize(new EntityStatus()
                {
                    EntityExists = this.context.State.EntityExists,
                    QueueSize = this.context.State.Queue?.Count ?? 0,
                    LockedBy = this.context.State.LockedBy,
                    CurrentOperation = opStatus,
                });
```
Most of the time these fields are uninteresting (because requests are typically processed instantly). But if operations have high latency or entities are locked one could potentially see that here, so it can help with troubleshooting.

I agree with @cgillum that the size of the current state, in bytes ,would also make sense. Perhaps we can add it to this same JSON object.

**Comment by ConnorMcMahon** [comment]
Pushing to our next release.

**Comment by cgillum** [comment]
Hey @davidmrdavid and/or @anthonychu can we consider this issue fixed based on the work done in #1301?

**Comment by davidmrdavid** [comment]
Yes, the work for this is done on our end! I think the portal feature remains hidden for internal testing only, but it's "out of our hands" for now :) 

Closing!
